util/worker.js

```diff
@@ -0,0 +1,291 @@
+import { EventEmitter } from "node:events";
+import PQueue from "p-queue";
+import puppeteer from "puppeteer-core";
+
+import { Logger } from "./logger.js";
+
+const logger = new Logger();
+
+export const CONCURRENCY_PAGE = "page";
```

Comment:
We can probably just get ride of concurrency options, and just always use the window concurrency, eg. open a new window

Reply:
Done